### ISSUE TITLE
cli: cdk migrate wrongly converts casing in ecs task definition props

### ISSUE THREAD

**Issue by tmokmss** [aws/aws-cdk#29027]
### Describe the bug

cdk migrate fails in handling the casing of ECS task definition properties. See the reproduction steps below.

### Expected Behavior

casing is kept from the original template.

```ts
          logConfiguration: {
            options: {
              awslogs-group: '/aws/ecs/test/main',
              awslogs-region: 'ap-northeast-1',
              awslogs-stream-prefix: 'ecs',
            },
            logDriver: 'awslogs',
          },
```

### Current Behavior

they're converted to camel case.

```ts
          logConfiguration: {
            secretOptions: [
            ],
            options: {
              awslogsGroup: '/aws/ecs/test/main',
              awslogsRegion: 'ap-northeast-1',
              awslogsStreamPrefix: 'ecs',
            },
            logDriver: 'awslogs',
          },
```

It results in the error below when you try to deploy the template:

> Resource handler returned message: "Invalid request provided: Create TaskDefinition: Log driver awslogs requires options: awslogs-region, awslogs-group (Service: AmazonECS; Status Code: 400
; Error Code: ClientException; ...

### Reproduction Steps

Deploy this template from CFn management console:

```yaml
AWSTemplateFormatVersion: '2010-09-09'

Resources:
  BackendECSTaskRole:
    Type: AWS::IAM::Role
    Properties:
      Path: /
      AssumeRolePolicyDocument:
        Statement:
          - Action: sts:AssumeRole
            Effect: Allow
            Principal:
              Service: ecs-tasks.amazonaws.com

  ECSTaskExecutionRole:
    Type: AWS::IAM::Role
    Properties:
      Path: /
      AssumeRolePolicyDocument:
        Statement:
          - Action: sts:AssumeRole
            Effect: Allow
            Principal:
              Service: ecs-tasks.amazonaws.com
      ManagedPolicyArns:
        - arn:aws:iam::aws:policy/service-role/AmazonECSTaskExecutionRolePolicy
        - arn:aws:iam::aws:policy/AmazonSSMReadOnlyAccess
        - arn:aws:iam::aws:policy/SecretsManagerReadWrite

  BackendServiceECSTaskDefinition:
    Type: AWS::ECS::TaskDefinition
    Properties:
      Family: test
      RequiresCompatibilities:
        - FARGATE
      Memory: "1024"
      Cpu: "256"
      NetworkMode: awsvpc
      ExecutionRoleArn: !GetAtt ECSTaskExecutionRole.Arn
      TaskRoleArn: !GetAtt BackendECSTaskRole.Arn
      ContainerDefinitions:
        - Name: "main"
          Image: "nginx"
          LogConfiguration:
            Options:
              awslogs-group: '/aws/ecs/test/main'
              awslogs-region: ap-northeast-1
              awslogs-stream-prefix: ecs
            LogDriver: awslogs
```

And then use cdk migrate:

```sh
npx cdk@latest migrate --from-stack --stack-name xxx
```

### Additional Information/Context

There are some props in CFn that casing should remain unchanged. One of them is an object of string that takes any string as a key: [example](https://docs.aws.amazon.com/ja_jp/AWSCloudFormation/latest/UserGuide/aws-properties-ecs-taskdefinition-logconfiguration.html#cfn-ecs-taskdefinition-logconfiguration-options)

### CDK CLI Version

2.126.0

### Node.js Version

v20.10.0

### OS

macOS

### Language

TypeScript

**Comment by TheRealAmazonKendra** [comment]
Duplicate-ish of https://github.com/aws/aws-cdk/issues/28997.  Same issue in the code, different manifestation of the bug.

**Comment by TheRealAmazonKendra** [comment]
This fix for this was released in v2.132.
